server/bin/pbench-index.py

```diff
@@ -468,23 +468,26 @@ def main(options, name):
                     rename_tb_link(
                         tb, Path(controller_path, f"{linkerrdest}.1"), idxctx.logger,
                     )
-                elif tb_res in (2, 3):
+                elif tb_res.value in (2, 3):
                     assert False, (
                         f"Logic Bomb!  Unexpected tar ball handling "
-                        f"result status {tb_res:d} for tar ball {tb}"
+                        f"result status {tb_res.value:d} for tar ball {tb}"
                     )
-                elif tb_res >= 4 or res < error_codes["GENERIC_ERROR"].value:
+                elif tb_res.value >= 4 or tb_res.value < 12:
```

Comment:
Now this really makes no sense; 12 is always `>= 4` and [0, 1, 2, 3] are `< 12` so I'm pretty sure this doesn't do whatever we want it to do. Should this be `and` so we ignore [4, 5, 6, 7, 8, 9] errors? That lines up with Pete's explanation earlier that we're ignoring the "tarball format" errors assuming they've already been logged.

So [1, 2, 3, 8, 9, 12] are _internal_ errors (that shouldn't or can't happen), while [4, 5, 6, 7] are errors in the tarball format. Maybe it makes sense to silently ignore tarballs that are clearly bad, while reporting the internal errors and potentially retrying. This seems to somewhat tie to the `and` alternative. If that was the intent, then I think the solution would be an Error object property defining each error as "internal" or "tarball_error", and then this would be `elif tb_res.tarball_error:`

Then again, the range includes "internal" indexing/template errors [8,9] which Pete suggests are simply included here because they're already logged. (Although I don't see any errors detected that _aren't_ logged above, so I'm not convinced that tracks.) We already assume this was broken originally in checking both `res` and `tb_res`, so it's not hard to believe that this should really be `>= 4 and < 8` if it comes to that.

And the way the data is structured now, the explicit `ignorable` property could be dropped and we could just have

```
def ignorable(self):
    return self.value == 0
```

Or maybe even call it _success_ to be more clear.